### PR TITLE
fix: overflow of the vtap_id value range

### PR DESCRIPTION
**Pull request by rosenlo**
### This PR is for:
- Server
- Agent

#### Affected branches
- main

### CODE REVIEW

File: agent/src/sender/uniform_sender.rs [hunk 1]
```diff
@@ -86,7 +86,7 @@ struct Header {
     version: u32, // 用来校验encode和decode是否配套
     team_id: u32,
     organize_id: u32,
-    vtap_id: u16, // roze用来上报server活跃的VTAP信息
+    vtap_id: u32, // roze用来上报server活跃的VTAP信息
```
Comment (yuanchaoa) [inline agent/src/sender/uniform_sender.rs:89]: 这里的修改会导致版本不兼容   @lzf575 也确认一下吧， 应该需要改下version ？

File: server/ingester/droplet/adapter/sequential_decoder.go [hunk 2]
```diff
@@ -371,7 +371,7 @@ func (d *SequentialDecoder) DecodeHeader(frameSize uint16) (bool, uint16) {
 	if version != _VERSION {
 		return true, 0
 	}
-	vtapId := d.U16()
+	vtapId := d.U32()
```
Comment (yuanchaoa) [inline server/ingester/droplet/adapter/sequential_decoder.go:374]: 这里应该做下版本兼容 ？ @lzf575

**Comment by sharang** [comment]
@rosenlo 
Changing to uint32 will introduce some compatibility issues. Additionally, there are still some agent_id-related contents on the server side (i.e., ClickHouse field) that haven't been modified in this PR. Does the number of agents in your environment exceed 64k?

**Comment by rosenlo** [comment]
> @rosenlo Changing to uint32 will introduce some compatibility issues. Additionally, there are still some agent_id-related contents on the server side (i.e., ClickHouse field) that haven't been modified in this PR. Does the number of agents in your environment exceed 64k?

Yes, this compatibility issue requires both agent and server to be updated, otherwise there will be errors such as the ```proto log decode failed``` (already encountered). In addition, because there are many modules involved, and some parts may be missing and need your supplement.

In our environment, the number of agents did not exceed 64k but due to multiple registrations and k8s node turnover, the agent_id had exceeded 100k.

**Comment by sharang** [comment]
> due to multiple registrations and k8s node turnover, the agent_id had exceeded 100k.

sounds like a bug in deepflow-server. @SongZhen0704

**Comment by rosenlo** [comment]
> > due to multiple registrations and k8s node turnover, the agent_id had exceeded 100k.
> 
> sounds like a bug in deepflow-server. @SongZhen0704

Sorry, the description of "multiple registrations" may be inappropriate, the fact that we manually deleted the agent registration data in MySQL during testing so when the agent registers again, the ID will increase.

**Comment by sharang** [comment]
As expected, when deepflow-server assigns agent_id, if it detects that the count exceeds 64K, it will attempt to recycle IDs starting from 1. Let's check if there are any issues in this part of the logic.

**Comment by sharang** [comment]
> > > due to multiple registrations and k8s node turnover, the agent_id had exceeded 100k.
> > 
> > 
> > sounds like a bug in deepflow-server. @SongZhen0704
> 
> Sorry, the description of "multiple registrations" may be inappropriate, the fact that we manually deleted the agent registration data in MySQL during testing so when the agent registers again, the ID will increase.

@SongZhen0704

**Comment by sharang** [comment]
@rosenlo We will fix the bug where agent_id exceeds 64K in the v6.3/v6.4/v6.5 branches. Subsequently, we will attempt to support a larger number of agents in the main branch based on this PR.

**Pull request closed without merging**